src/Controllers/QuestionTemplateController.php

```diff
@@ -1,68 +1,118 @@
 use QuizApp\Services\QuizTemplateServices;
+use Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException;
 
+/**
+ * Class QuestionTemplateController
+ * @package QuizApp\Controllers
+ */
 class QuestionTemplateController extends AbstractController
 {
 
+    /**
+     * @var QuestionTemplateServices
+     */
     private $questionTemplateServices;
 
+    /**
+     * @var Paginator
+     */
+    private $paginator;
+
     /**
      * UserController constructor.
      * @param RendererInterface $renderer
      * @param QuestionTemplateServices $questionInstanceServices
+     * @param Paginator $paginator
      */
-    public function __construct(RendererInterface $renderer, QuestionTemplateServices $questionInstanceServices)
+    public function __construct(RendererInterface $renderer, QuestionTemplateServices $questionInstanceServices, Paginator $paginator)
     {
         parent::__construct($renderer);
         $this->questionTemplateServices = $questionInstanceServices;
+        $this->paginator = $paginator;
     }
 
+    /**
+     * This method creates a user and saves it in the database
+     *
+     * @param Request $request
+     * @return Response
+     */
     public function createQuestion(Request $request)
     {
         $text = $request->getParameter('text');
         $type = $request->getParameter('type');
         $answer = $request->getParameter('answer');
-        $this->questionTemplateServices->saveQuestion($text, $type,$answer);
+        $this->questionTemplateServices->saveQuestion($text, $type, $answer);
         $location = 'Location: http://quizApp.com/admin-questions-listing?page=1';
         $body = Stream::createFromString("");
 
         return new Response($body, '1.1', '301', $location);
     }
 
+    /**
+     * This method returns a Response with the questions from a specified page and text
+     *
+     * @param Request $request
+     * @param array $requestAttributes
+     * @return Response
+     */
     public function showQuestions(Request $request, array $requestAttributes)
     {
-        $arguments['currentPage'] = (int)$request->getParameter('page');
-        $arguments['pages'] = $this->questionTemplateServices->getQuestionNumber($requestAttributes);
+        $text = $request->getParameter('text');
+        $currentPage = (int)$request->getParameter('page');
+        $totalResults = (int)$this->questionTemplateServices->getQuestionNumberOfPagesByText($text);
+
         $arguments['username'] = $this->questionTemplateServices->getName();
-        $arguments['questions'] = $this->questionTemplateServices->getQuestions($requestAttributes, $request->getParameter('page'));
+        $arguments['text'] = $text;
+        $arguments['questions'] = $this->questionTemplateServices->getQuestionsByText($text, $request->getParameter('page'));
+
+        //sets the currentPage, totalResults and totalPages to us them in the view from the paginator variable
```

Comment:
This comment describes the code exactly :) A comment should add new information / describe the high-level intention of a line of code.